expertsystem/io/dot.py

```diff
@@ -30,55 +32,88 @@ def write(instance: object, filename: str) -> None:
         stream.write(output_str)
 
 
-_DOT_HEAD = """
-digraph {
+_DOT_HEAD = """digraph {
     rankdir=LR;
-    node [shape=point];
-    edge [arrowhead=none, labelfloat=true];
-    """
+    node [shape=point, width=0];
+    edge [arrowhead=none];
+"""
 _DOT_TAIL = "}\n"
 _DOT_RANK_SAME = "    {{ rank=same {} }};\n"
 _DOT_DEFAULT_NODE = '    "{}" [shape=none, label="{}"];\n'
 _DOT_DEFAULT_EDGE = '    "{}" -> "{}";\n'
 _DOT_LABEL_EDGE = '    "{}" -> "{}" [label="{}"];\n'
 
 
-def __graph_to_dot(graph: StateTransitionGraph) -> str:
-    def node_name(edge_id: int, node_id: Optional[int] = None) -> str:
-        if node_id is None:
-            return f"edge{edge_id}"
-        return f"node{node_id}"
+def __graph_list_to_dot(graphs: List[StateTransitionGraph]) -> str:
+    dot_source = _DOT_HEAD
+    for i, graph in enumerate(graphs):
+        dot_source += __graph_to_dot_content(graph, prefix=f"g{i}_")
+    dot_source += _DOT_TAIL
+    return dot_source
 
-    def format_particle(node_edge_ids: List[int]) -> str:
-        name_list = [f'"{node_name(i)}"' for i in node_edge_ids]
-        return ",".join(name_list)
 
+def __graph_to_dot(graph: StateTransitionGraph) -> str:
     dot_source = _DOT_HEAD
+    dot_source += __graph_to_dot_content(graph)
+    dot_source += _DOT_TAIL
+    return dot_source
 
+
+def __graph_to_dot_content(
+    graph: StateTransitionGraph, prefix: str = ""
+) -> str:
+    dot_source = ""
     top = graph.get_initial_state_edges()
     outs = graph.get_final_state_edges()
-    for i in top:
-        dot_source += _DOT_DEFAULT_NODE.format(
-            node_name(i), graph.edge_props[i]["Name"]
-        )
-    for i in outs:
+    for edge_id in top + outs:
         dot_source += _DOT_DEFAULT_NODE.format(
-            node_name(i), graph.edge_props[i]["Name"]
+            prefix + __node_name(edge_id), __edge_label(graph, edge_id)
```

Comment:
why are nodes labeled?

Reply:
The problem is that DOT language only works with nodes. So actually the node names are confusing: we prepend `"edge"` to a node that represents the end point of an open edge (initial and final state) and prepend `"node"` to internal nodes. See output of `test_dot.py`:

```dot
digraph {
    rankdir=LR;
    node [shape=point, width=0];
    edge [arrowhead=none];
    "edge0" [shape=none, label="J/psi(1S)[1]"];
    "edge2" [shape=none, label="gamma[1]"];
    "edge3" [shape=none, label="pi0[0]"];
    "edge4" [shape=none, label="pi0[0]"];
    { rank=same "edge0" };
    { rank=same "edge2", "edge3", "edge4" };
    "edge0" -> "node0";
    "node0" -> "node1" [label="f(0)(980)[0]"];
    "node0" -> "edge2";
    "node1" -> "edge3";
    "node1" -> "edge4";
}
```